packages/bee-q/src/components/avatar/bq-avatar.tsx

```diff
@@ -98,43 +110,72 @@ export class BqAvatar {
+        this.trimmedInitials = this.initials.substring(0, this.getIndex(size));
+      }
+    });
+  };
+
+  private getIndex = (size: TAvatarSize): number => {
+    switch (size) {
+      case 'small':
+        return 2;
+      case 'medium':
+        return 3;
+      case 'large':
+        return 4;
+      default:
+        // also if size === xsmall
+        return 1;
+    }
+  };
+
   // render() function
   // Always the last one in the class.
   // ===================================
 
   render() {
     return (
-      <div
-        class={{
-          'relative overflow-hidden bg-ui-secondary-light': true,
-          [`size--${this.size}`]: true,
-          'rounded-full': this.shape === 'circle',
-          'rounded-xs': this.shape === 'square' && this.size === 'xsmall',
-          'rounded-s': this.shape === 'square' && this.size === 'small',
-          'rounded-m': this.shape === 'square' && (this.size === 'medium' || this.size === 'large'),
-        }}
-        aria-label={this.label}
-        role="img"
-        part="base"
-      >
-        {this.initials && (
-          <span
-            class="absolute left-0 top-0 inline-flex h-full w-full items-center justify-center font-bold"
-            part="text"
-          >
-            {this.initials}
-          </span>
-        )}
-        {this.image && !this.hasError && (
-          <img
-            class="absolute left-0 top-0 h-full w-full object-cover"
-            alt={this.altText ?? undefined}
-            src={this.image}
-            onError={this.onImageError}
-            part="img"
-          />
+      <Host>
+        <div
+          class={{
+            'relative overflow-hidden border-[2px] border-solid border-stroke-tiertary bg-ui-secondary-light': true,
+            [`size--${this.size}`]: true,
+            'rounded-full': this.shape === 'circle',
+            'rounded-xs': this.shape === 'square' && this.size === 'xsmall',
+            'rounded-s': this.shape === 'square' && this.size === 'small',
+            'rounded-m': this.shape === 'square' && (this.size === 'medium' || this.size === 'large'),
+          }}
+          aria-label={this.label}
+          role="img"
+          part="base"
+        >
+          {this.initials && (
+            <span
+              class="absolute left-0 top-0 inline-flex h-full w-full items-center justify-center font-bold"
+              part="text"
+            >
+              {this.trimmedInitials}
+            </span>
+          )}
+          {this.image && !this.hasError && (
+            <img
+              class="absolute left-0 top-0 h-full w-full object-cover"
+              alt={this.altText ?? undefined}
+              src={this.image}
+              onError={this.onImageError}
+              part="img"
+            />
+          )}
+        </div>
+        {this.badgeContent && (
+          <bq-badge size="small" text-color="#fff">
+            {this.badgeContent}
+          </bq-badge>
```

Comment:
I think we need a slot instead, the idea is to allow the user to pass the badge on integration, now through a property value.